tests/unit_tests/test_launch/test_project/test_project.py

```diff
@@ -199,6 +199,23 @@ def mock_project_args():
     }
 
 
+def test_project_parse_existing_requirements_invalid_requirement(
+    mocker, tmp_path, mock_project_args
+):
+    mocker.logger = MagicMock()
+    mocker.patch("wandb.sdk.launch._project_spec._logger", mocker.logger)
```

Comment:
Use the [wandb_caplog fixture](https://github.com/wandb/wandb/blob/2d31a4bc90780f2314b66c33d822975398a33e91/tests/conftest.py#L147-L163) (used in the same way as pytest's [caplog](https://docs.pytest.org/en/stable/how-to/logging.html#caplog-fixture))